// the data structures that passe between nodes
package PS.CXXNET;
import "data/proto/data.proto";

// the configuration to start a node
message Config {
  // a worker will report its progress to the scheduler every _report_interval_
  // second
  optional int32 report_interval = 4 [default = 1];

  optional bool pressure_test = 10;

}

// the progress a worker/server reports to the scheduler
message Progress {
  repeated double objective = 1;
  optional uint64 num_examples_processed = 2;
  repeated double accuracy = 3;
}

// the commands the scheduler issues to worker/servers
message Call {
  enum Command {
    UPDATE_MODEL = 1;
    REPORT_PROGRESS = 2;
    SAVE_MODEL = 3;
    RECOVER = 4;
    COMPUTE_VALIDATION_AUC = 5;
  }
  required Command cmd = 1;
  optional DataConfig data = 2;
}


message PressureTest {
  repeated int32 key = 1;
  repeated int32 size = 2;
  optional int32 max_delay = 3 [default = 0];
}
